src/react-imgix.jsx

```diff
@@ -109,68 +110,97 @@ function buildSrc({
     fixedSize && width ? { width } : {}
   );
 
-  const src = constructUrl(rawSrc, srcOptions);
+  if (disableSrcSet) {
+    const src = constructUrl(rawSrc, srcOptions);
+    return { src, src };
+  }
 
-  let srcSet;
+  if (fixedSize) {
+    const { q, ...urlParams } = srcOptions;
+    const params = {};
 
-  if (disableSrcSet) {
-    srcSet = src;
-  } else {
-    if (fixedSize) {
-      const { q, ...urlParams } = srcOptions;
-      const constructedUrl = constructUrl(rawSrc, urlParams);
-
-      let srcFn = buildDprSrcWithQualityByDpr;
-      if (q) {
-        srcFn = buildDprSrcWithQuality;
-      } else if (disableQualityByDPR) {
-        srcFn = buildDprSrcWithoutQuality;
+    for (const k in urlParams) {
+      if (PARAM_EXPANSION[k]) {
+        params[PARAM_EXPANSION[k]] = urlParams[k];
+      } else {
+        params[k] = urlParams[k];
       }
+    }
```

Comment:
Might be out of scope here but just a general thought: Would it be worth packaging this functionality up in some sort of helper, since it appears here as well: https://github.com/imgix/react-imgix/pull/776/files#diff-11b2734ddfc432c9651ade2ada0deb5f2201e6e88e3fcc271ebbcac05de8c82fR118-R124

Reply:
Definitely, I think I should do that next. Right now this pattern exists in three different places, I'll wrap it up into a helper.